lib/sinatra/base.rb

```diff
@@ -1343,19 +1343,19 @@ def mime_types(type)
       # Define a before filter; runs before all requests within the same
       # context as route handlers and may access/modify the request and
       # response.
-      def before(path = /.*/, **options, &block)
+      def before(path = /(.*)/, **options, &block)
         add_filter(:before, path, options, &block)
       end
 
       # Define an after filter; runs after all requests within the same
       # context as route handlers and may access/modify the request and
       # response.
-      def after(path = /.*/, **options, &block)
+      def after(path = /(.*)/, **options, &block)
         add_filter(:after, path, options, &block)
       end
 
       # add a filter
-      def add_filter(type, path = /.*/, **options, &block)
+      def add_filter(type, path, **options, &block)
```

Comment:
I think this method's argument of `path` doesn't need default value

Reply:
Potentially breaking compatibility, I don't think there is a reason for breaking it.